pandas/errors/__init__.py

```diff
@@ -157,6 +157,13 @@ class MergeError(ValueError):
     """
 
 
+class SortError(TypeError):
+    """
+    Error raised when problems arise during sorting due to problems
```

Comment:
Is it possible to be more explicit about the types of "problems" that could arise? Is this just when sorting mixed types?

Reply:
@WillAyd : the rationale here was to a create an exception analogous to the custom MergeError. Exceptions should be considered part of the api and so this SortError is a subclass of TypeError to be backwards compatible with the exceptions currently raised by np.sort from safe_sort and therefore result in a non-breaking api change.

Rather than be explicit on the exceptions covered, I consider this to be a base class for any sort related errors that may need to be handled. (although i would probably not have chosen TypeError as the base class). I should probably subclass this again for the safe_sort errors, but that is not strictly necessary at the moment.